BLSQ/iaso
# Polio-1253: Refactor campaigns front-end

Related JIRA tickets :POLIO-1253

## Self proofreading checklist


## Changes

- convert `Dashboard` to tsx
- remove modals from `Dashboard`'s jsx
- remove useless state from Dashboard.tsx
- move table columns code to separate hook
- convert `CreateEditDialog` to tsx, move tabs code into a hook for legibility
- rename `PageActionWithLink` as `LinkButton`
- remove `ConfirmDialog`, `PageActions` and `PageAction` components
- replace `PageActions` with `DashboardButtons`
- move edit, delete and restore dialogs in table columns hook
- change parameters of `useGetCampaigns` for smoother user experience
- improve loading state of the Create and Edit modals to show spinner while saving
- improve loading state of the table to show spinner when deleting/restoring a campaign
- align csv button with table and filter button

## How to test

Go to campaigns page, try functionalities. Nothing should have changed
Go to grouped campaigns and try outthe link. You should always have the "edit" modal open

## Print screen / video

![Screenshot 2023-12-21 at 15 41 20](https://github.com/BLSQ/iaso/assets/38907762/f3aac089-9592-44e5-aa5f-6b53fa6eaf97)

## Notes

The initial ticket was about links to campaigns that would sometimes open the create modal i.o the edit modal of the selected campaign. The refactor relies only on url params to open the edit modal (no internal state), so it should be fixed. It's also hard to test since the initial bug is kinda flaky

Needs to be merged after https://github.com/BLSQ/iaso/pull/989

DO NOT MERGE UNTIL CYPRESS TESTS ARE WRITTEN FOR THIS PAGE
